src/OtcOption.sol

```diff
@@ -0,0 +1,252 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+import "@chainlink/contracts/src/v0.8/interfaces/AggregatorV3Interface.sol";
+
+error Option__InvalidUnderlyingToken();
+error Option__InvalidQuoteToken();
+error Option__InvalidStrike();
+error Option__InvalidMaturity();
+error Option__InvalidAmount();
+error Option__InvalidPremium();
+
+error Option__PriceFeedNotFound();
+error Option__DepositPremiumFailed();
+error Option__DepositMarginFailed();
+error Option__TransferPremiumFailed();
+error Option__TransferMarginFailed();
+error Option__DealNotOpen();
+error Option__DealExpired();
+error Option__DealNotExpired();
+error Option__DealNotFound();
+error Option__DealHasNoTaker();
+
+contract OtcOption is Ownable {
+    event DealCreated(
+        uint id,
+        address underlyingToken,
+        address quoteToken,
+        uint strike,
+        uint maturity,
+        bool isCall,
+        uint amount,
+        uint premium,
+        bool isMakerBuyer,
+        address maker
+    );
+    event DealTaken(uint id, address taker);
+    event DealRemoved(uint id);
+    event DealSettled(uint id, bool exercised);
+
+    enum DealStatus {
+        Open,
+        Taken,
+        Removed,
+        Settled
+    }
+
+    struct Deal {
+        uint id;
+        address underlyingToken;
+        address quoteToken;
+        uint strike;
+        uint maturity;
+        bool isCall;
+        uint amount;
+        uint premium;
+        bool isMakerBuyer;
+        DealStatus status;
+        address maker;
+        address taker;
+    }
+
+    // status variables
+    mapping(address => mapping(address => address)) private _priceFeeds; // underlying => (quote => feed)
+    uint private _dealCounter = 1;
```

Comment:
Why not uint private _dealCounter; //So it's initialized to 0?

Reply:
I initialize to 1 because in the code I assume that a deal ID equal to 0 means that does not exist. I could also initialize this to 0 and then increment immediately when the first deal is created, but I thought this would be more explicit.

Reply:
Totally fine, just a design decision!